crates/ruff_linter/src/rules/refurb/rules/slice_to_remove_prefix_or_suffix.rs

```diff
@@ -334,8 +334,10 @@ fn affix_matches_slice_bound(data: &RemoveAffixData, semantic: &SemanticModel) -
             }),
         ) => num
             .as_int()
-            .and_then(ast::Int::as_u32) // Only support prefix removal for size at most `u32::MAX`
-            .is_some_and(|x| x == string_val.to_str().text_len().to_u32()),
+            // Only support prefix removal for size at most `u32::MAX`
+            .and_then(ast::Int::as_u32)
+            .and_then(|x| usize::try_from(x).ok())
```

Comment:
I suggest converting to a `u64` considering that you have to use `usize::try_from` anyways (for 32 bit platforms)
```suggestion
            .and_then(ast::Int::as_u64)
            .and_then(|x| usize::try_from(x).ok())
```

Reply:
Or you could consider adding a `as_usize` method to `ast::Int`

Reply:
went with the latter